### PR TITLE
add base::Time::monotonic()

### PR DESCRIPTION
**Pull request by doudou**
Meant for timeout calculations.

### CODE REVIEW

File: src/Time.cpp [hunk 1]
```diff
@@ -31,6 +34,14 @@ Time Time::now()
     return Time(static_cast<int64_t>(t.tv_sec) * UsecPerSec + t.tv_usec);
 }
 
+static auto monotonicClock = std::chrono::steady_clock();
```
Comment (2maz) [inline src/Time.cpp:37]: Why don't you keep monotonicClock inside the function scope of Time::monotonic? It is not used anywhere else.

Comment (doudou) [reply src/Time.cpp:37]: Because then initialization would not be thread safe (clock.now() is thread safe in C++11)

Comment (2maz) [reply src/Time.cpp:37]: According to the standard 11 draft: http://www.open-std.org/JTC1/SC22/WG21/docs/papers/2011/n3242.pdf
6.7.4:

"If control enters the declaration concurrently while the variable is being initialized, the concurrent execution shall wait for
completion of the initialization."

(see also https://iamroman.org/blog/2017/04/cpp11-static-init/)

Related to this for gcc: 
```
 -fno-threadsafe-statics
           Do not emit the extra code to use the routines specified in the C++
           ABI for thread-safe initialization of local statics.  You can use
           this option to reduce code size slightly in code that doesn't need
           to be thread-safe.

Comment (doudou) [reply src/Time.cpp:37]: Cool. Did not know that C++11 fixed that. Will change.

Comment (doudou) [reply src/Time.cpp]: Fixed.

**Review by 2maz — APPROVED** [review]
I suggest to remove the using statement in L14 and just stick to std::chrono in L43 - for the rest you used std:: anyhow.

**Comment by doudou** [comment]
I would rather add another commit to remove all the `std::` stuff ...

**Comment by 2maz** [comment]
I leave the choice to you, and at the same time add another option, how about adding:
using std::chrono within the scope of Time::monotonic?

**Comment by doudou** [comment]
I meant "for the whole file". See 9077b5e.

**Comment by 2maz** [comment]
> I meant "for the whole file". See [9077b5e](https://github.com/rock-core/base-types/commit/9077b5ec00b477bea77483a5a7367e3934491d4d).

I feared that ;) - but your choice, so accepted (review dialogue does not show for unknown reasons)

**Pull request merged**